src/pkg/p2p/preheat/provider/dragonfly.go

```diff
@@ -87,9 +87,17 @@ func (dd *DragonflyDriver) Preheat(preheatingImage *PreheatImage) (*PreheatingSt
 		return nil, err
 	}
 
+	// Currently if resource was preheated before at least once, Dragonfly will return StatusAlreadyReported.
```

Comment:
@yunkunrao 

Why don't directly check if the status == StatusAlreadyReported? I'm thinking the `result.ID == ""` may also happen in other cases.